spinalcordtoolbox/image.py

```diff
@@ -1704,3 +1707,66 @@ def compute_cross_corr_3d(image: Image, coord, xrange=list(range(-10, 10)), xshi
     # Change adjust rl_coord
     logger.info('R-L coordinate adjusted from %s to  %s)', x, x + xrange[ind_peak])
     return x + xrange[ind_peak]
+
+
+def stitch_images(fnames_in: list, fname_out='stitched.nii.gz'):
+    """
+    Stitch two (or more) images utilizing the C++-precompiled binaries of Biomedia-MIRA's stitching toolkit
+    (https://github.com/biomedia-mira/stitching) by placing a system call.
+
+    :param fnames_in: list of filenames of scans in RPI orientation. e.g. [stack1.nii.gz, stack2.nii.gz]
+    :param fname_out: filename for stitched output scan.
+    :param directory: path to temp directory where we store the result of stitching.nii.gz
+    :return: none
+    """
+    # stringify the fname list to parsable cmd parameter
+    fnames_cmd = " ".join(fnames_in)
+
+    cmd = ['isct_stitching', '-i', fnames_cmd, '-o', fname_out, '-a']
+    status, output = run_proc(cmd, verbose='verbose', is_sct_binary=True)
+    print(status, output)
+    if status != 0:
+        raise RuntimeError(f"Subprocess call {cmd} returned non-zero: {output}")
+
+    return Image(fname_out)
+
+
+def generate_stitched_qc_images(ims_in: Sequence[Image], im_out: Image):
+    """
+    Pad input and output images, so that a QC report can compare between the output of
+    'stitching', vs. the naive output of concatenating the input images together.
+    """
```

Comment:
I'm not sure where this "generate images for QC" functionality should be? I considered a few options:

1. `qc.py`
    
    For context, `qc.py` has the (very large) class [`QcImage`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/464721865ed5f7adbf6a755f8619816c75e8ef75/spinalcordtoolbox/reports/qc.py#L27-L515), which contains many methods for preparing input images for display, e.g.:
    
    https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/464721865ed5f7adbf6a755f8619816c75e8ef75/spinalcordtoolbox/reports/qc.py#L128-L137
    
    But, these methods in `QCImage` are mainly concerned with plotting behavior (e.g. matplotlib's axes, colormaps, drawing lines and shapes, etc.). In other words, they work to display existing images, whereas my `generate_stitched_qc_images` function modifies the images themselves using padding, concatenation, etc. So, I thought it might be a poor fit.

   _((Aside: The names and docstrings for some of the `QcImage` methods are quite confusing. I had to use a method called `no_seg_seg` to produce the stitching QC report. So, the QC module feels like a good candidate for some small documentation refactoring to make things a little more clear.))_

2. Creating a new API module, `stitching.py`, for both functions

   I figure that the stitching code is unlikely to be re-used outside of anything but a dedicated stitching script, so putting both functions in `image.py` might be bloating up that API module, since it (mostly) contains more generalized/reusable image processing functionality.

3. Keeping both functions in `image.py`

    `sct_image` <=> `image.py`, basically.

I went with 3. because it was the simplest option, given the existing state of this PR.

That said, API organization is something I'm not set in stone on? ((My goal is to balance making small, surgical changes to the existing legacy API vs. making useful reorganizations. Finding that middle-ground can be tricky sometimes.)) 

So, I would love to hear the opinion of e.g. @mguaypaq here.